serializer/serializer.c

```diff
@@ -673,6 +673,102 @@ status_t receive_mam_message_res_serialize(char* const message, char** obj) {
   return ret;
 }
 
+status_t send_mam_req_deserialize(const char* const obj, ta_send_mam_req_t* req) {
+  if (obj == NULL) {
+    ta_log_error("%s\n", "SC_SERIALIZER_NULL");
+    return SC_SERIALIZER_NULL;
+  }
+  cJSON* json_obj = cJSON_Parse(obj);
+  cJSON* json_result = NULL;
+  status_t ret = SC_OK;
+
+  if (json_obj == NULL) {
+    ret = SC_SERIALIZER_JSON_PARSE;
+    ta_log_error("%s\n", "SC_SERIALIZER_JSON_PARSE");
+    goto done;
+  }
+
+  json_result = cJSON_GetObjectItemCaseSensitive(json_obj, "seed");
+  if (json_result != NULL) {
+    size_t seed_size = strlen(json_result->valuestring);
```

Comment:
Since seed is fixed size, we can use `strnlen` instead of `strlen` here.

Reply:
I just noticed we have a function `ta_json_get_string`, we can use it here.

Reply:
I found some problem that these fileds are optional, so if I use `ta_json_get_string`, then I will do some work on determing whether this field is empty or not